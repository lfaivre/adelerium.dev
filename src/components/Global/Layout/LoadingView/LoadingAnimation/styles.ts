import tw, { styled, TwStyle } from 'twin.macro';

// @todo Move color map to shared utility

type LoadingAnimationBorderColorValues = `border-offwhite` | `border-charcoal` | `border-offpink`;

const loadingAnimationBorderColorMap: { [key in LoadingAnimationBorderColorValues]: TwStyle } = {
  'border-offwhite': tw`border-offwhite`,
  'border-charcoal': tw`border-charcoal`,
  'border-offpink': tw`border-offpink`,
};

type LoadingAnimationProps = { borderColor: LoadingAnimationBorderColorValues };

export const LoadingAnimationWrapper = styled.div<LoadingAnimationProps>`
  position: relative;
  width: 5rem;
  height: 5rem;

  & div {
    position: absolute;
    opacity: 1;
    border-width: 0.125rem;
    border-style: solid;
    border-radius: 50%;
    ${({ borderColor }) => loadingAnimationBorderColorMap[borderColor]}
    animation: ripple 1.5s cubic-bezier(0, 0.2, 0.8, 1) infinite;
  }

  & div:nth-of-type(2) {
    animation-delay: -0.75s;
  }

  @keyframes ripple {
    0% {
      top: 2rem;
      left: 2rem;
      opacity: 1;
      width: 1rem;
      height: 1rem;
    }
    100% {
      top: 0;
      left: 0;
      opacity: 0;
      width: 5rem;
      height: 5rem;
    }
  }
`;
