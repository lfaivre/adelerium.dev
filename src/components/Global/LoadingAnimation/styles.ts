import { styled } from 'twin.macro';

export const LoadingAnimationContainer = styled.div`
  position: relative;
  width: 5rem;
  height: 5rem;

  & div {
    position: absolute;
    border: 0.25rem solid var(--color-OffPink);
    border-radius: 50%;
    opacity: 1;
    animation: ripple 1.5s cubic-bezier(0, 0.2, 0.8, 1) infinite;
  }

  & div:nth-of-type(2) {
    animation-delay: -0.75s;
  }

  @keyframes ripple {
    0% {
      top: 2rem;
      left: 2rem;
      width: 1rem;
      height: 1rem;
      opacity: 1;
    }
    100% {
      top: 0;
      left: 0;
      width: 5rem;
      height: 5rem;
      opacity: 0;
    }
  }
`;
