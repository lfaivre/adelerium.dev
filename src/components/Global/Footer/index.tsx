import { useFooterQueryData } from '@adelerium/components/Global/Footer/useFooterQueryData';
import { StyledInternalLink } from '@adelerium/components/Global/StyledInternalLink';
import { Next, Previous } from '@adelerium/constants/presentation';
import { studioCopyrightText, studioUrl, websiteFullPath } from '@adelerium/constants/site-metadata';
import { useAppState } from '@adelerium/hooks/app-state';
import { usePathData } from '@adelerium/hooks/usePathData';
import { BoldParagraphType, BoldType, BrandingType, NormalParagraphType } from '@adelerium/styles/text';
import { FlexColumnWrapper, FlexRowWrapper } from '@adelerium/styles/wrappers';
import { getRandomInt } from '@adelerium/utils/math';
import { OutboundLink } from 'gatsby-plugin-google-analytics';
import React, { ReactElement } from 'react';
import tw, { css } from 'twin.macro';

const factOnError = `This site was built with Gatsby.js.`;
const staticStudioCTA = `Need a website?`;
const staticStudioUrlText = `kevaladesign.com`;
const staticStudioLogoText = `KD.`;
const staticFactTitle = `Did you know?`;
const staticLinkedInText = `li.`;
const staticGitHubText = `gh.`;

export const Footer = (): ReactElement => {
  const { footerData, brandingLink, linkedInLink, gitHubLink } = useFooterQueryData();

  const {
    theme: { colors },
  } = useAppState();
  const { pathname, isIndex, pathData, isValidPath } = usePathData();

  const getRandomFact = (): string => {
    if (!footerData?.facts) return factOnError;
    return footerData.facts[getRandomInt(0, footerData.facts.length - 1)]?.text || factOnError;
  };

  return (
    <div tw="px-4 pt-8 pb-4 md:p-8 w-full">
      <FlexRowWrapper alignItems="items-start" justifyContent="justify-start" tw="hidden md:flex mb-8 w-full">
        <FlexColumnWrapper alignItems="items-start" justifyContent="justify-start" tw="w-2/6">
          <BoldParagraphType color={colors.secondary.default} defaultFontSize>
            {staticStudioCTA}
          </BoldParagraphType>
          <OutboundLink
            href={brandingLink?.destination || websiteFullPath}
            label={brandingLink?.destination || websiteFullPath}
            target="_blank"
            rel="noopener noreferrer"
          >
            <NormalParagraphType color={colors.secondary.default} defaultFontSize>
              {staticStudioUrlText}
            </NormalParagraphType>
          </OutboundLink>
        </FlexColumnWrapper>
        <FlexColumnWrapper alignItems="items-center" justifyContent="justify-start" tw="w-2/6">
          <BoldParagraphType color={colors.secondary.default} defaultFontSize textAlign="text-center">
            {staticFactTitle}
          </BoldParagraphType>
          <NormalParagraphType color={colors.secondary.default} defaultFontSize textAlign="text-center" tw="w-full">
            {getRandomFact()}
          </NormalParagraphType>
        </FlexColumnWrapper>
        <FlexRowWrapper alignItems="items-start" justifyContent="justify-end" tw="w-2/6">
          <OutboundLink
            href={linkedInLink?.destination || websiteFullPath}
            label={linkedInLink?.destination || websiteFullPath}
            target="_blank"
            rel="noopener noreferrer"
            tw="mr-4"
          >
            <BoldParagraphType color={colors.secondary.default} defaultFontSize>
              {staticLinkedInText}
            </BoldParagraphType>
          </OutboundLink>
          <OutboundLink
            href={gitHubLink?.destination || websiteFullPath}
            label={gitHubLink?.destination || websiteFullPath}
            target="_blank"
            rel="noopener noreferrer"
          >
            <BoldParagraphType color={colors.secondary.default} defaultFontSize>
              {staticGitHubText}
            </BoldParagraphType>
          </OutboundLink>
        </FlexRowWrapper>
      </FlexRowWrapper>
      <FlexRowWrapper alignItems="items-start" justifyContent="justify-start" tw="mb-8 w-full">
        <FlexRowWrapper alignItems="items-start" justifyContent="justify-start" tw="w-1/2 md:w-2/6">
          {pathData && (
            <StyledInternalLink
              pathname={pathname}
              isIndex={isIndex}
              pathData={pathData}
              isValidPath={isValidPath}
              direction={Previous}
            />
          )}
        </FlexRowWrapper>
        <FlexRowWrapper alignItems="items-start" justifyContent="justify-center" tw="hidden md:flex w-2/6">
          <OutboundLink
            href={brandingLink?.destination || studioUrl}
            label={brandingLink?.destination || studioUrl}
            target="_blank"
            rel="noopener noreferrer"
          >
            <BrandingType color={colors.secondary.default} defaultFontSize>
              {staticStudioLogoText}
            </BrandingType>
          </OutboundLink>
        </FlexRowWrapper>
        <FlexRowWrapper alignItems="items-start" justifyContent="justify-end" tw="w-1/2 md:w-2/6">
          {pathData && (
            <StyledInternalLink
              pathname={pathname}
              isIndex={isIndex}
              pathData={pathData}
              isValidPath={isValidPath}
              direction={Next}
            />
          )}
        </FlexRowWrapper>
      </FlexRowWrapper>
      <hr
        css={[
          css`
            border-color: ${colors.secondary.default};
          `,
          tw`block md:hidden mb-4 md:mb-2 border-t w-full h-0`,
        ]}
      />
      <FlexRowWrapper alignItems="items-center" justifyContent="justify-start" tw="w-full">
        <FlexRowWrapper alignItems="items-start" justifyContent="justify-center" tw="w-full overflow-hidden">
          <BoldType
            color={colors.secondary.default}
            textAlign="text-center"
            wordBreak="break-normal"
            tw="w-full uppercase text-xs"
          >
            {studioCopyrightText}
          </BoldType>
        </FlexRowWrapper>
      </FlexRowWrapper>
    </div>
  );
};
