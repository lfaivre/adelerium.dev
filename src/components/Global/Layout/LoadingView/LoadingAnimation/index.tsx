import React, { ReactElement } from 'react';

import { LoadingAnimationWrapper } from './styles';

export const LoadingAnimation = (): ReactElement => {
  return (
    <LoadingAnimationWrapper borderColor="border-offpink">
      <div />
      <div />
    </LoadingAnimationWrapper>
  );
};
