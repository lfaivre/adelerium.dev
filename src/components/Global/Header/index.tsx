import { StyledInternalLink } from '@adelerium/components/Global/StyledInternalLink';
import { homePageTitleText, notFoundPageTitleText } from '@adelerium/constants/paths';
import { Next } from '@adelerium/constants/presentation';
import { useAppDispatch, useAppState } from '@adelerium/hooks/app-state';
import { SET_VIEW } from '@adelerium/hooks/app-state/constants';
import { usePathData } from '@adelerium/hooks/usePathData';
import { BoldParagraphType, BoldTypeAsButton } from '@adelerium/styles/text';
import { FlexRowWrapper } from '@adelerium/styles/wrappers';
import React, { ReactElement, useEffect, useState } from 'react';
import tw from 'twin.macro';

const defaultHeaderText = `Welcome`;

type HeaderProps = { disableToggle?: boolean };

export const Header = ({ disableToggle }: HeaderProps): ReactElement => {
  const {
    view: {
      sideBar: { isVisible: sideBarIsVisible },
    },
  } = useAppState();
  const dispatch = useAppDispatch();
  const { pathname, isIndex, pathData, isValidPath } = usePathData();
  const [headerTitle, setHeaderTitle] = useState<string>(defaultHeaderText);
  const [toggleIsVisible, setToggleIsVisible] = useState<boolean>(false);

  useEffect(() => {
    const { text } = pathData || { text: notFoundPageTitleText };
    const newHeaderTitle = text !== homePageTitleText ? text : defaultHeaderText;
    setHeaderTitle(newHeaderTitle);
  }, [pathData]);

  return (
    <FlexRowWrapper alignItems="items-center" justifyContent="justify-between" tw="w-full h-full">
      <FlexRowWrapper
        alignItems="items-center"
        justifyContent="justify-start"
        onMouseOver={() => !disableToggle && setToggleIsVisible(true)}
        onMouseOut={() => setToggleIsVisible(false)}
        tw="relative w-1/2 h-full"
      >
        <BoldParagraphType
          color="text-charcoal"
          textAlign="text-left"
          css={[
            tw`z-0 transition-opacity duration-300 ease-in-out text-3xl md:text-4xl`,
            toggleIsVisible && tw`opacity-0`,
          ]}
          aria-label="Toggle Side Bar Navigation"
        >{`${headerTitle}.`}</BoldParagraphType>
        <BoldTypeAsButton
          disabled={disableToggle}
          color="text-charcoal"
          textAlign="text-left"
          onClick={() =>
            !disableToggle && dispatch({ type: SET_VIEW, payload: { sideBar: { isVisible: !sideBarIsVisible } } })
          }
          css={[
            tw`block absolute left-0 z-0 transition-opacity duration-300 ease-in-out opacity-0 focus:outline-none pt-1 md:pt-2 h-full uppercase`,
            toggleIsVisible && tw`opacity-100`,
            disableToggle && tw`cursor-default`,
          ]}
        >
          Toggle Navigation
        </BoldTypeAsButton>
      </FlexRowWrapper>
      <FlexRowWrapper alignItems="items-center" justifyContent="justify-end" tw="w-1/2 h-full">
        <StyledInternalLink
          pathname={pathname}
          isIndex={isIndex}
          pathData={pathData}
          isValidPath={isValidPath}
          direction={Next}
        />
      </FlexRowWrapper>
    </FlexRowWrapper>
  );
};
