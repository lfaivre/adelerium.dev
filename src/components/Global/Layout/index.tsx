import React, { ReactElement, useEffect, useRef } from 'react';
import { GlobalStyles } from 'twin.macro';

import { useAppState, useAppDispatch } from '../../../shared/hooks/global-state';
import { SET_LOADING, SET_LAYOUT_WIDTH } from '../../../shared/types/state';

import { DefaultView } from './DefaultView';
import { LoadingView } from './LoadingView';

import { FlexColumnWrapper } from '../../../shared/styles/wrappers';

type LayoutProps = { children: ReactElement };

export const Layout = ({ children }: LayoutProps): ReactElement => {
  const { isLoading, windowWidth } = useAppState();
  const dispatch = useAppDispatch();

  const layoutRef = useRef<HTMLDivElement | null>(null);

  useEffect(() => {
    dispatch({ type: SET_LOADING, isLoading: false });
  }, [dispatch]);

  useEffect(() => {
    if (layoutRef.current && layoutRef.current.clientWidth) {
      const { clientWidth: layoutWidth } = layoutRef.current;
      dispatch({ type: SET_LAYOUT_WIDTH, layoutWidth });
    }
  }, [windowWidth, dispatch]);

  return (
    <>
      <GlobalStyles />
      <FlexColumnWrapper
        alignItems="items-center"
        justifyContent="justify-start"
        ref={layoutRef}
        tw="relative w-full max-w-global h-screen overflow-hidden"
      >
        {isLoading && <LoadingView />}
        <DefaultView>{children}</DefaultView>
      </FlexColumnWrapper>
    </>
  );
};

type PageWrapperElementProps = { element: ReactElement; props: Record<string, unknown> };

export const PageWrapperElement = ({ element, props }: PageWrapperElementProps): ReactElement => {
  // eslint-disable-next-line react/jsx-props-no-spreading
  return <Layout {...props}>{element}</Layout>;
};
