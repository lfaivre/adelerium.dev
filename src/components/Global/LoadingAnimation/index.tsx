import { LoadingAnimationWrapper } from '@adelerium/components/Global/LoadingAnimation/styles';
import React, { ReactElement } from 'react';

export const LoadingAnimation = (): ReactElement => {
  return (
    <LoadingAnimationWrapper borderColor="border-charcoal">
      <div />
      <div />
    </LoadingAnimationWrapper>
  );
};
