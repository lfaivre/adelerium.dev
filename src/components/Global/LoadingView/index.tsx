import React from 'react';
import { LoadingAnimation } from '../LoadingAnimation';

import { LoadingViewContainer } from './styles';

export const LoadingView = (): JSX.Element => {
  return (
    <LoadingViewContainer>
      <LoadingAnimation />
    </LoadingViewContainer>
  );
};
