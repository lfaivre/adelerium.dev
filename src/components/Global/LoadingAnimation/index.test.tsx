import { LoadingAnimation } from '@adelerium/components/Global/LoadingAnimation';
import React from 'react';
import renderer from 'react-test-renderer';

describe(`LoadingAnimation`, () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<LoadingAnimation />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
