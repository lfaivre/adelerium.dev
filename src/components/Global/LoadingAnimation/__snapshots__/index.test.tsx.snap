// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAnimation renders correctly 1`] = `
<div
  className="css-1yf9vtf"
>
  <div />
  <div />
</div>
`;
